mlrun/config.py

```diff
@@ -328,6 +352,8 @@ def _do_populate(env=None):
     del config._cfg["kfp_image"]
     config._cfg["_dask_kfp_image"] = config._cfg["dask_kfp_image"]
     del config._cfg["dask_kfp_image"]
+    config._cfg["_iguazio_api_url"] = config._cfg["iguazio_api_url"]
+    del config._cfg["iguazio_api_url"]
```

Comment:
why is this being done? guessing there's a good reason but the existing HACK comment isn't clear
are the `_X` keys handled differently in some other place?

Reply:
We just need to have a key that is different than the property name (otherwise you're going recursive, improved the comment there